libs/api/domains/rsk-company-info/src/lib/graphql/models/rskCompanySearchItems.model.ts

```diff
@@ -0,0 +1,24 @@
+import { Field, ObjectType, ID } from '@nestjs/graphql'
+import { RskCompanyLink } from './rskCompanyLink.model'
+import { RskCompanySearchItem } from './rskCompanySearchItem.model'
+
+@ObjectType()
+export class RskCompanySearchItems {
+  @Field(() => [RskCompanySearchItem])
+  items?: RskCompanySearchItem[]
```

Comment:
Should this not return `RskCompany`. This would help with updating the Apollo cache in the front-end.

Reply:
I was wondering about this too.

Should we do N+1 queries to get full company information in search results? Or should we have a `RskCompanySearchItem` with basic details and a `searchItem.company` field resolver which returns the full company details?

Reply:
We could extract the difference into a new field with its own field resolver (info).
This means that fetching the default search items only uses a single query, but if more detailed result is needed for the array, then n+1 calls will be made.